types/firestore.d.ts

```diff
@@ -582,6 +582,28 @@ declare namespace FirebaseFirestore {
       ...fieldsOrPrecondition: any[]
     ): Promise<WriteResult>;
 
+    /**
+     * Attaches a listener that is run every time a BulkWriter operation fails.
+     *
+     * @param errorFn A callback to be called every time a BulkWriter operation
+     * fails.
+     */
+    onError(
+      errorFn: (error: Error, operation: BulkWriterOperation) => void
+    ): void;
+
+    /**
+     * Retry a BulkWriter operation that has failed.
+     *
+     * The BulkWriterOperation object is a parameter on the `onError()`
+     * callback function that is called whenever a write fails.
```

Comment:
We should mention that is can only be used in combination with `flush()`. This should probably also be part of the error message when a retry is enqueue after `close()` is called.